lightbar_manager/src/lightbar_manager/lightbar_manager_worker.cpp

```diff
@@ -75,8 +75,34 @@ namespace lightbar_manager
         return;
     }
 
+    std::vector<lightbar_manager::LightBarIndicator> LightBarManagerWorker::handleTurnSignal(const automotive_platform_msgs::TurnSignalCommandPtr& msg_ptr)
+    {
+        std::vector<lightbar_manager::LightBarIndicator> turn_signal;
+        if (msg_ptr->turn_signal == current_turn_signal_)
+        {
+            return {};
+        }
+        if (msg_ptr->turn_signal == automotive_platform_msgs::TurnSignalCommand::LEFT) //NONE -> LEFT
+        {
+            turn_signal.push_back(lightbar_manager::LightBarIndicator::YELLOW_ARROW_LEFT);  
+        }
+        else if (msg_ptr->turn_signal == automotive_platform_msgs::TurnSignalCommand::RIGHT) //NONE -> RIGHT
+        {
+            turn_signal.push_back(lightbar_manager::LightBarIndicator::YELLOW_ARROW_RIGHT);
+        }
+        else if (msg_ptr->turn_signal == automotive_platform_msgs::TurnSignalCommand::NONE) 
+        {
+            // check previous signal
+            if (current_turn_signal_ == automotive_platform_msgs::TurnSignalCommand::RIGHT) // RIGHT -> NONE
```

Comment:
The comments day Right-> None and Left->None but it looks like the code is doing Right->Right and Left->Left?

Reply:
This is because that logic hits when the light is instructed to turn off. So if the prev state was left, it should still return left so that the next logic knows which light to turn off